OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -209,6 +209,7 @@ public MovementType CurrentMovementTypes
 		public bool IsBlocking { get; private set; }
 
 		public bool IsMovingBetweenCells => FromCell != ToCell;
+		public MoveResult MoveResult;
```

Comment:
I feel like we're abusing `Mobile` to store this state. I think I'd prefer if `Move` stored the `MoveResult` and we used a pattern like this in the calling activities.

```c#
Move lastMove;

public override bool Tick(Actor self)
{
    ...
    // Our move is already completed, either we got to the destination or we got blocked along the way.
    if (lastMove?.MoveResult.IsComplete())
        return true;

   ...
   // Time to make a move.
   lastMove = move.MoveOfSomeKind(...);
   QueueChild(lastMove);

   ...
}